### PR TITLE
Use the built-in cmake python module

### PR DESCRIPTION
**Pull request by bramtayl**
Python has a built-in python find module that does a lot of the work for us

Also added a mini-macro for evaluating boolean expressions for convenience

### CODE REVIEW

File: Python/CMakeLists.txt [hunk 1]
```diff
@@ -1,22 +1,14 @@
 ## Python module ##
 
-option(BUILD_PYTHON_INTERFACE "Build the Python interface" ON)
+option(INSTALL_PYTHON_INTERFACE "Build the Python interface" ON)
```
Comment (stekyne) [inline Python/CMakeLists.txt:3]: Can't really rename this as it would break the public API that people are using

File: Python/CMakeLists.txt [hunk 2]
```diff
@@ -1,22 +1,14 @@
...
-      SET(PYTHON3_MODULE_INSTALL_DIR ${PYTHON3_SITE_PACKAGES} CACHE
-       PATH "Python3 module install dir")
-       
-      if (PYTHON3_MODULE_INSTALL_DIR)
-         message(STATUS "PYTHON3 MODULE INSTALL DIR: ${PYTHON3_MODULE_INSTALL_DIR}" )
-        install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/ctcsound.py
-            DESTINATION ${PYTHON3_MODULE_INSTALL_DIR})
-      endif()
+assign_bool(NOT_APPLE NOT APPLE)
+check_deps(INSTALL_PYTHON_INTERFACE NOT_APPLE Python3_FOUND)
```
Comment (stekyne) [inline Python/CMakeLists.txt:8]: Why not on APPLE? The existing code doesn't seem to make that distinction?

File: Python/CMakeLists.txt (hunk 1 above)
Comment (bramtayl) [reply Python/CMakeLists.txt:3]: Well, I added the option in https://github.com/csound/csound/pull/1744 two months ago, so I doubt anyone is relying on it, but happy to use the original name

File: Python/CMakeLists.txt (hunk 2 above)
Comment (bramtayl) [reply Python/CMakeLists.txt:8]: Well, there is a `AND NOT APPLE` in the original code, I have no idea why it's there though. We could potentially try to get rid of it and see what happens?

**Pull request merged**